### ISSUE TITLE
✨ Making experimental factory functions of `NotEmptySet.Companion` available on Java

### ISSUE THREAD

**Issue by LVMVRQUXL** [kotools/types#511]
## 📝 Description

We would like to make the **experimental** factory functions of the `NotEmptySet.Companion` type with the following signature available for Java users:

```kotlin
fun <E> NotEmptySet.Companion.create(collection: Collection<E>): NotEmptySet<E>
fun <E> NotEmptySet.Companion.createOrNull(collection: Collection<E>): NotEmptySet<E>?

fun <E> NotEmptySet.Companion.of(head: E, vararg tail: E): NotEmptySet<E>
```

This means that its compatibility should be tested and its documentation should indicate their usage from Java code.

## 🔗 Dependencies

This issue is blocked by the following ones:
- [x] #353 
- [x] #436 

## ✅ Checklist

- [ ] Make the `create(Collection)` function available for Java users, [test] its compatibility, update the public [API binaries] and update the [unreleased changelog].
- [ ] Make the `createOrNull(Collection)` function available for Java users, [test] its compatibility, update the public [API binaries] and update the [unreleased changelog].
- [ ] Make the `of(E, vararg E)` function available for Java users, [test] its compatibility, update the public [API binaries] and update the [unreleased changelog].
- [ ] Close this issue as completed and update tracking ones if present.

[api binaries]: https://github.com/kotools/types/blob/main/CONTRIBUTING.md#checking-the-api-binaries
[test]: https://github.com/kotools/types/blob/main/CONTRIBUTING.md#running-tests
[unreleased changelog]: https://github.com/kotools/types/blob/main/CHANGELOG.md#unreleased

**Comment by LVMVRQUXL** [comment]
Another issue for stabilizing these functions will be created while resolving #619.
